### PR TITLE
Add test for setting table header depth

### PR DESCRIPTION
**Pull request by julieg18**
Followup to #2517

### CODE REVIEW

File: webview/src/experiments/components/table/Table.test.tsx [hunk 1]
```diff
@@ -258,6 +258,23 @@ describe('Table', () => {
     })
   })
 
+  describe('Head Depth', () => {
+    it('should be updated by the user in the header context menu', async () => {
```
Comment (julieg18) [inline webview/src/experiments/components/table/Table.test.tsx:262]: Do we need more than this test for this feature?

Comment (mattseddon) [reply webview/src/experiments/components/table/Table.test.tsx:262]: "Backend" -> https://github.com/iterative/vscode-dvc/blob/558c09e09947ef755067aec509e7fefafb4aa4ba/extension/src/test/suite/experiments/index.test.ts#L172

**Review by sroy3 — APPROVED** [review]

File: extension/src/test/suite/experiments/index.test.ts [hunk 2]
```diff
@@ -878,6 +878,29 @@ suite('Experiments Test Suite', () => {
...
+
+      const messageReceived = new Promise(resolve =>
+        disposable.track(mockMessageReceived.event(() => resolve(undefined)))
+      )
+
+      mockMessageReceived.fire({
+        type: MessageFromWebviewType.SET_EXPERIMENTS_HEADER_DEPTH
+      })
+
+      expect(executeCommandStub).to.be.calledWith(
```
Comment (julieg18) [inline extension/src/test/suite/experiments/index.test.ts:897]: While creating this test, I realized that we don't have a telemetry event set up for this command. Will add in a followup!

Comment (sroy3) [reply extension/src/test/suite/experiments/index.test.ts:897]: Seems like that test is actually failing

Comment (julieg18) [reply extension/src/test/suite/experiments/index.test.ts:897]: Figured it out! I now realize I have completely misunderstood how the end to end tests work and need to rewrite this. Converting to a draft for now!

File: extension/src/test/suite/experiments/index.test.ts [hunk 3]
```diff
@@ -878,6 +878,26 @@ suite('Experiments Test Suite', () => {
...
+
+      const mockMessageReceived = getMessageReceivedEmitter(webview)
+      const messageReceived = new Promise(resolve =>
+        disposable.track(mockMessageReceived.event(() => resolve(undefined)))
+      )
+      mockMessageReceived.fire({
+        type: MessageFromWebviewType.SET_EXPERIMENTS_HEADER_DEPTH
+      })
+
+      expect(mockShowInputBox).to.be.calledOnce
```
[the remark this answers is not in the document]
Comment (mattseddon) [reply extension/src/test/suite/experiments/index.test.ts:896]: You can use this: https://github.com/iterative/vscode-dvc/blob/58d85c0593524968974b61c06725bc7cc4a607cd/extension/src/test/suite/util.ts#L42-L43

[the remark this answers is not in the document]
Comment (julieg18) [reply extension/src/test/suite/experiments/index.test.ts]: Thanks! Also found a util for the `inputBox`! Any issues with the test that I missed?

**Review by mattseddon — APPROVED** [review]

**Pull request merged**